### PR TITLE
HDDS-9704. Eliminate dependency on spotbugs-annotations

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

* Remove usage of `@SuppressFBWarnings`
* Remove dependency on `spotbugs-annotations`
* Fix SpotBugs violations or suppress in `findbugsExcludeFile.xml`
* Remove instructions on how to suppress warnings from `CONTRIBUTING.md`.  I don't think it's appropriate for new contributors.

See separate commits for details.

https://issues.apache.org/jira/browse/HDDS-9704

## How was this patch tested?

CI:
https://github.com/adoroszlai/ozone/actions/runs/6892845091

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -159,18 +159,6 @@ Some scripts require third-party tools, but most of these are installed during t
 
 Most scripts (except `build.sh`) output results in `target/<name>`, e.g. `target/docs`.
 
-### False positive findbugs violation
```
Comment (jojochuang) [inline CONTRIBUTING.md:162]: Maybe leave this line and add instructions for using dev-support/findbugsExcludeFile.xml file to exclude false positives?

**Review by jojochuang — COMMENTED** [review]
Would be nice to also forbid spotbugs-annotations from future inclusions as well.

**Comment by adoroszlai** [comment]
> Would be nice to also forbid spotbugs-annotations from future inclusions as well.

Filed HDDS-9703 for that.

File: CONTRIBUTING.md (hunk 1 above)
Comment (adoroszlai) [reply CONTRIBUTING.md:162]: I don't think this level of detail belongs to this guide, intended for new contributors.

The example given in the guide was actually a bug, shouldn't have been suppressed.

**Review by xBis7 — APPROVED** [review]
@adoroszlai Thanks for the cleanup and the improvements. LGTM!

**Review by hemantk-12 — COMMENTED** [review]
Thanks for the cleanup @adoroszlai 

Overall looks good to me.

I've a quick question:
Why are some of the source (e.g. RandomKeyGenerator) and test (e.g. TestServerNotLeaderExceptionMessageParsing) files not added to `findbugsExcludeFile.xml`?

**Comment by adoroszlai** [comment]
> Why are some of the source (e.g. RandomKeyGenerator) and test (e.g. TestServerNotLeaderExceptionMessageParsing) files not added to `findbugsExcludeFile.xml`?

The warning(s) in those files are fixed.

**Review by hemantk-12 — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @hemantk-12, @jojochuang, @xBis7 for the review.

**Pull request merged**